.github/workflows/build.yml

```diff
@@ -253,7 +253,6 @@ jobs:
             - uses: actions/setup-python@v5
               with:
                 python-version: '3.11'
-            - uses: lukka/get-cmake@latest
```

Comment:
Realistically the version of CMake provided by the runner images is always pretty new and gets updated frequently.

The main benefit of lukka/get-cmake was
1.) Test older version of cmake
2.) Get ninja and cache it.

Reason 2 is now outdated for most of these checks. I recommend removing it for that reason. Saves a little bit of CI cache on runs that don't need it.